examples/plot_03_competing_risks.py

```diff
@@ -0,0 +1,250 @@
+#
+# - It is a generalization of the accuracy metric in the survival and the competing
+#   risks setting, which is the proportion of correct predictions (e.g. highest
+#   predicted probability) at a fixed time.
+# - It is computed for different time horizons, given by the user, with direct time
+#   stamps or quantiles of the observed durations.
+# - For a given patient at a fixed time, we compare the actual observed event to the
+#   most predicted one. The censored patients are removed from the computation after
+#   their censoring duration. In other word, let's imagine a patient who has
+#   experienced death by cancer at time :math:`t`. Before this time, we want the model
+#   to predict with the highest probability that this patient will survive. After
+#   :math:`t`, we want the model to predict the death by cancer event with the highest
+#   probability.
+# - The mathematical formula is:
+#
+# .. math::
+#     \mathrm{acc}(\zeta) = \frac{1}{n_{nc}} \sum_{i=1}^n ~ I\{\hat{y}_i=y_{i,\zeta}\}
+#        \overline{I\{\delta_i = 0 \cap t_i \leq \zeta \}}
+#
+# where:
+#
+# - :math:`\zeta` is a fixed time horizon.
+# - :math:`n_{nc}` is the number of uncensored individuals at :math:`\zeta`.
+# - :math:`\delta_i` is the event experienced by the individual :math:`i` at
+#   :math:`t_i`.
+# - :math:`\hat{y} = \text{arg}\max\limits_{k \in [0, K]} \hat{F}_k(\zeta|X=x_i)`
+#   where :math:`\hat{F}_0(\zeta|X=x_i) \triangleq \hat{S}(\zeta|X=x_i)`.
+#   :math:`\hat{y}` is the most probable predicted event for individual :math:`i`
+#   at :math:`\zeta`.
+# - :math:`y_{i,\zeta} = \delta_i ~ I\{t_i \leq \zeta \}` is the observed event
+#   for individual :math:`i` at :math:`\zeta`.
+#
+#      **References:**
+#
+#      .. [Alberge2024] J. Alberge, V. Maladiere,  O. Grisel, J. Abécassis,
+#      G. Varoquaux, "Survival Models: Proper Scoring Rule and Stochastic Optimization
+#      with Competing Risks", 2024
+
+# %%
+# Usage
+# =====
+#
+# Generating synthetic data
+# -------------------------
+#
+# We begin by generating a linear, synthetic dataset. For each individual, we uniformly
+# sample a shape and scale value, which we use to parameterize a Weibull distribution,
+# from which we sample a duration.
+from hazardous.data import make_synthetic_competing_weibull
+from sklearn.model_selection import train_test_split
+
+
+X, y = make_synthetic_competing_weibull(n_events=3, n_samples=10_000, return_X_y=True)
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+X_train.shape, y_train.shape
+
+# %%
+# Next, we display the distribution of our target.
+import seaborn as sns
+from matplotlib import pyplot as plt
+
+
+sns.histplot(
+    y_test,
+    x="duration",
+    hue="event",
+    multiple="stack",
+    palette="colorblind",
+)
+plt.show()
+
+# %%
+# Computing the Accuracy in Time
+# -------------------------------------------
+#
+# After training SurvivalBoost, we compute its accuracy in time.
```

Comment:
maybe add a sentence here about the quantiles "we compute its accuracy in time at 16 quantiles corresponding to the proportion of censored individuals" (maybe you have a clearer sentence, but this quantile thing has never felt obvious to me)